shotover-proxy/src/transforms/cassandra/sink_cluster/node_pool.rs

```diff
@@ -138,16 +143,20 @@ impl NodePool {
         // TODO this should use the keyspace info to properly select the replica count
         let replica_host_ids = self
             .token_map
-            .iter_replica_nodes(Murmur3Token::generate(&routing_key), 1);
+            .iter_replica_nodes(Murmur3Token::generate(&routing_key), rf)
+            .collect::<Vec<uuid::Uuid>>();
+
+        let (dc_replicas, rack_replicas) = self
+            .nodes
+            .iter_mut()
+            .filter(|node| replica_host_ids.contains(&node.host_id) && node.is_up)
+            .split(|node| node.rack == rack);
```

Comment:
Not sure if pulling in a dependency just for this is worth it. (not asking you to change it, just saying I'm unsure)
But just in case you didnt know the two split halves have an internal allocation each.

Reply:
I did attempt to solve this without the dep, but it was challenging to get around the borrow check for the mutable reference to the nodes vector.

Reply:
I gave it a quick go and wow, thats super weird, the borrow checker is getting caught on something even when I clearly separate the lifetimes.
The split iterator nicely works around whatever the problem is so fine by me :+1: